src/compiler/ast/doc/JSDocTypeExpression.ts

```diff
@@ -0,0 +1,14 @@
+import { ts } from "../../../typescript";
+import { Node } from "..";
+
+/**
+ * JS doc type expression node
+ */
+export class JSDocTypeExpression extends Node<ts.JSDocTypeExpression> {
+    /**
+     * Gets the type node of the JS Doc type expression
+     */
+    getTypeNode() {
+        return this.compilerNode.type;
```

Comment:
This will return the typescript compiler api node instead of the wrapped one. Use either `this._getNodeFromCompilerNode` or `this._getNodeFromCompilerNodeIfExists` (depending on whether `type` is nullable) similar to what you did in `JSDocPropertyLikeTag`.

Hmmm... I should actually combine `_getNodeFromCompilerNode` and `_getNodeFromCompilerNodeIfExists` sometime, but that's for another day.